patient-db/src/main/java/com/hartwig/hmftools/patientdb/clinical/ClinicalAlgo.java

```diff
@@ -154,48 +121,60 @@ private static List<Patient> readEcrfPatients(@NotNull EcrfPatientReader reader,
     }
 
     @NotNull
-    private List<Patient> readWidePatients(@NotNull Map<String, List<SampleData>> samplesPerPatient,
-            @NotNull Map<String, ConsentConfig> consentConfigMap) {
+    private List<Patient> readLimsPatients(@NotNull Map<String, List<SampleData>> samplesPerPatient,
+            Map<String, ConsentConfig> consentConfigMap, List<DoidNode> doidNodes) {
         List<Patient> patients = Lists.newArrayList();
+        CorePatientReader corePatientReader = new CorePatientReader(primaryTumorCurator);
+        LimsPatientReader limsPatientReader = new LimsPatientReader(doidNodes);
 
-        WidePatientReader widePatientReader = new WidePatientReader(ecrfModels.wideModel(), primaryTumorCurator, treatmentCurator);
         for (Map.Entry<String, List<SampleData>> entry : samplesPerPatient.entrySet()) {
             List<SampleData> tumorSamples = tumorSamplesOnly(entry.getValue());
-            if (!tumorSamples.isEmpty() && tumorSamples.get(0).cohortId().equals("WIDE")) {
+            if (!tumorSamples.isEmpty()) {
                 String patientId = entry.getKey();
+
                 // We assume every sample for a single patient has the same primary tumor.
-                String primaryTumor = tumorSamples.get(0).limsPrimaryTumor();
-                patients.add(widePatientReader.read(patientId,
-                        primaryTumor,
-                        sequencedSamplesOnly(tumorSamples),
-                        consentConfigMap,
-                        tumorSamples.get(0).cohortId()));
+                SampleData chosenSample = tumorSamples.get(0);
+                String logInfo = join("|", chosenSample.sampleBarcode(), chosenSample.sampleId(),
+                        chosenSample.cohortId());
+
+                if (!hasAllRequiredFields(chosenSample)) {
+                    LOGGER.warn("Not processing sample from LIMS/LAMA due to required fields missing [{}]", logInfo);
+                } else if (hasLamaBasedCuration(chosenSample)) {
+                    LOGGER.info("Processing sample from LIMS/LAMA sample with curation from LAMA [{}]", logInfo);
+                    patients.add(limsPatientReader.read(patientId, chosenSample, sequencedSamplesOnly(tumorSamples)));
+                } else if (isCuratedFromEcrfSource(chosenSample)) {
```

Comment:
Is this intentional: If the sample has a "lims tumor location" it will have already been processed in the previous else if case. This one will just never be executed.

Reply:
This is intentional indeed and stems from the fact that only newer samples have their curation in LAMA, but might also have the "lims tumor location" regardless.